### PR TITLE
🐛 bugfix function aggregateFromMachinesToKCP

### PR DESCRIPTION
**Pull request by wm775825**
**What this PR does / why we need it**:

When aggregating from machine conditions to kcp condition, the priority is:
- false conditions with error reason
- false conditions with warning reason
- false conditions with info reason
- true conditions
- unknown conditions

However, when dealing with "false conditions with info conditions", there is a bug in source codes.

In Line 562, having checked warning reasons, we should check info reasons next. That is, to check "kcpMachinesWithInfo" rather than "kcpMachinesWithWarnings". However, there is a duplicate "kcpMachinesWithWarnings" in line 556 and line 662.

Maybe this is not a common scene, so it didn't lead to serious problems and was neglected. 

As a beginner of cluster-api, i found it when i read source codes, so i fix it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

no issues.

### CODE REVIEW

**Comment by wm775825** [comment]
how to invoke lint? anyone can help?

**Comment by killianmuldoon** [comment]
/ok-to-test

(Not sure why lint isn't running) :thinking:

**Comment by oscr** [comment]
In the warning it says: "This workflow requires approval from a maintainer." So I assume I'm not allowed to trigger it for none Kubernetes org members.

**Comment by sbueringer** [comment]
It cannot be done via ok-to-test (I think there is a long-standing issue for that in test-infra)

Someone with at least write access to the repo has to push a button

**Review by g-gaston — COMMENTED** [review]
The fix looks good, thanks for finding this!
Could we add a unit test to make sure this doesn't break again?

**Comment by wm775825** [comment]
i will add a unit test

**Review by vincepri — APPROVED** [review]
/lgtm
/assign @fabriziopandini

**Comment by jackfrancis** [comment]
Thanks for this fix, @wm775825!

**Comment by jackfrancis** [comment]
@vincepri @sbueringer should we cherry pick this back to release-1.2 and release-1.3?

**Comment by wm775825** [comment]
sorry, it seems that i removed some reviewers by mistake?

**Comment by sbueringer** [comment]
@jackfrancis I'll try to take a look soon'ish. If it's a bugfix and pretty safe would be good to cherry-pick in both

**Comment by sbueringer** [comment]
Thank you very much!!

/lgtm
/approve

**Comment by sbueringer** [comment]
@wm775825 Do you have time to open (manual) cherry-pick PRs for release-1.3 and release-1.2?

**Comment by wm775825** [comment]
it seems there has been two manual cherrypicks merged.

**Comment by jackfrancis** [comment]
@wm775825 I went ahead and took the liberty, thanks again for doing all the real work here :) 💪

**Pull request merged**
